addons/hide-ads/addon.json

```diff
@@ -0,0 +1,60 @@
+{
+  "name": "Hide advertising comments",
+  "description": "Automatically hides comments that contain certain strings deemed as advertising.",
+  "tags": ["comments", "community"],
+  "versionAdded": "1.37.0",
+  "credits": [{ "name": "Jazza", "link": "http://scratch.mit.edu/users/greeny--231" }],
+  "userscripts": [{ "matches": ["profiles", "projects", "studios"], "url": "userscript.js" }],
+  "userstyles": [{ "matches": ["profiles", "projects", "studios"], "url": "userstyle.css" }],
+  "settings": [
+    { "name": "Project comments", "type": "boolean", "default": true, "id": "projects" },
+    { "name": "Studio comments", "type": "boolean", "default": true, "id": "studios" },
+    { "name": "Profile comments", "type": "boolean", "default": true, "id": "profiles" },
+    {
+      "name": "Method of hiding advertising",
+      "type": "select",
+      "potentialValues": [
+        { "id": "censor", "name": "Censor content" },
+        { "id": "hide", "name": "Hide comment" }
+      ],
+      "id": "method",
+      "default": "censor"
+    },
+    { "name": "Apply general spam filter", "type": "boolean", "default": true, "id": "spamFilter" },
+    {
+      "name": "Strings that match as advertising",
+      "id": "matches",
+      "type": "table",
+      "default": [{ "matchType": "link", "matchValue": "scratch.mit.edu/projects" }],
+      "row": [
+        {
+          "name": "Type",
+          "id": "matchType",
+          "type": "select",
+          "default": "link",
+          "potentialValues": [
+            { "name": "Link", "id": "link" },
+            { "name": "String", "id": "string" },
+            { "name": "User", "id": "user" },
+            { "name": "RegExp", "id": "regExp" }
```

Comment:
passing regexps are very unsafe. some regexps can cause the page to freeze up if you run them on some strings.

a malicious actor could send a user a setting file containing an unsafe regexp, then get them to visit a certain comment, causing their browser to freeze, which is less than ideal

of course this isn't the worst thing that could happen, but IMO we just shouldn't add a regex filter option.

Reply:
that is a good point, plus im sure no one would use it anyway, i use some built in regex anyway